src/python/pants/backend/project_info/tasks/export_dep_as_jar.py

```diff
@@ -0,0 +1,301 @@
+# Copyright 2014 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+import json
+import os
+import zipfile
+from collections import defaultdict
+
+from twitter.common.collections import OrderedSet
+
+from pants.backend.jvm.subsystems.jvm_platform import JvmPlatform
+from pants.backend.jvm.subsystems.scala_platform import ScalaPlatform
+from pants.backend.jvm.targets.jar_library import JarLibrary
+from pants.backend.jvm.targets.junit_tests import JUnitTests
+from pants.backend.jvm.targets.jvm_target import JvmTarget
+from pants.backend.jvm.targets.scala_library import ScalaLibrary
+from pants.backend.project_info.tasks.export import SourceRootTypes
+from pants.backend.project_info.tasks.export_version import DEFAULT_EXPORT_VERSION
+from pants.base.build_environment import get_buildroot
+from pants.build_graph.resources import Resources
+from pants.java.distribution.distribution import DistributionLocator
+from pants.java.jar.jar_dependency_utils import M2Coordinate
+from pants.task.console_task import ConsoleTask
+from pants.util.contextutil import temporary_file
+from pants.util.memo import memoized_property
+
+
+class ExportDepAsJar(ConsoleTask):
+  """[Experimental] Create project info for IntelliJ with dependencies treated as jars.
+
+  This is an experimental task that mimics export but uses the jars for
+  jvm dependencies instead of sources.
+  """
+
+  @classmethod
+  def register_options(cls, register):
+    super().register_options(register)
+    register('--formatted', type=bool, implicit_value=False,
+      help='Causes output to be a single line of JSON.')
+
+  @classmethod
+  def prepare(cls, options, round_manager):
+    round_manager.require_data('runtime_classpath')
```

Comment:
I think we might need to do a `super().prepare(options, round_manager)` here too?

Reply:
Good catch! Corrected.